patches/react-native-onyx+1.0.15.patch

```diff
@@ -0,0 +1,175 @@
+         // Object values are merged one after the other
+-        return _.reduce(mergeValues, (modifiedData, mergeValue) => {
+-            const newData = lodashMerge({}, modifiedData, mergeValue);
++        const before = performance.now();
++        const res =  _.reduce(mergeValues, (modifiedData, mergeValue) => {
++            const newData = Object.assign({}, lodashMerge(modifiedData, mergeValue));
+ 
+             // We will also delete any object keys that are undefined or null.
+             // Deleting keys is not supported by AsyncStorage so we do it this way.
+             // Remove all first level keys that are explicitly set to null.
+             return _.omit(newData, (value, finalObjectKey) => _.isNull(mergeValue[finalObjectKey]));
+         }, data || {});
++        const after = performance.now();
++        console.log('merge took', after - before, "for key", key);
++        return res;
+     }
+ 
+     // If we have anything else we can't merge it so we'll
+diff --git a/node_modules/react-native-onyx/lib/OnyxCache.js b/node_modules/react-native-onyx/lib/OnyxCache.js
+index 287bc3b..ffc3faa 100644
+--- a/node_modules/react-native-onyx/lib/OnyxCache.js
++++ b/node_modules/react-native-onyx/lib/OnyxCache.js
+@@ -1,5 +1,6 @@
+ import _ from 'underscore';
+-import lodashMerge from 'lodash/merge';
++//import lodashMerge from 'lodash/merge';
++import { merge as lodashMerge } from './fastMerge';
+ 
+ const isDefined = _.negate(_.isUndefined);
+ 
+@@ -110,7 +111,10 @@ class OnyxCache {
+      * @param {Record<string, *>} data - a map of (cache) key - values
+      */
+     merge(data) {
+-        this.storageMap = lodashMerge({}, this.storageMap, data);
++        const before = performance.now();
++        this.storageMap = Object.assign({}, lodashMerge(this.storageMap, data));
++        const after = performance.now();
++        console.log("aaa merge cache took", after - before);
+ 
+         const storageKeys = this.getAllKeys();
+         const mergedKeys = _.keys(data);
+diff --git a/node_modules/react-native-onyx/lib/fastMerge.js b/node_modules/react-native-onyx/lib/fastMerge.js
+new file mode 100644
+index 0000000..f3adbc8
+--- /dev/null
++++ b/node_modules/react-native-onyx/lib/fastMerge.js
+@@ -0,0 +1,97 @@
++/**
++ * Is the object Mergeable
++ *
++ * @param val
++ * @returns {*|boolean}
++ */
++ function isMergeableObject(val) {
++  var nonNullObject = val && typeof val === 'object';
++  return nonNullObject &&
++      Object.prototype.toString.call(val) !== '[object RegExp]' &&
++      Object.prototype.toString.call(val) !== '[object Date]';
++}
++/**
++* Empty the Target
++*
++* @param val
++* @returns {*}
++*/
++function emptyTarget(val) {
++  return Array.isArray(val) ? [] : {}
++}
++/**
++* Clone if Necessary
++*
++* @param value
++* @param optionsArgument
++* @returns {*}
++*/
++function cloneIfNecessary(value, optionsArgument) {
++  var clone = optionsArgument && optionsArgument.clone === true;
```

Comment:
Where does the `.clone` argument come from? Is that really needed? I don't think it's something we use today

Reply:
The whole code is mostly taken from here https://medium.com/@lubaka.a/how-to-remove-lodash-performance-improvement-b306669ad0e1 That's why it's here